packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -909,17 +891,100 @@ export const useArbTokenBridge = (
       decimals,
       listIds: new Set()
     }
-
     setBridgeTokens(oldBridgeTokens => {
       return { ...oldBridgeTokens, ...bridgeTokensToAdd }
     })
-
     updateErc20L1Balance([l1AddressLowerCased])
     if (l2Address) {
       updateErc20L2Balance([l2Address])
     }
   }
 
+  async function addToken(erc20L1orL2Address: string) {
+    // 1. first check where the token exists - on l1 or l2
+    // if found on l1, then extract its l2 and add it. simple. (addValidBridgeableToken(l1Address, l2Address))
+    // (not else) if found on l2, there are 2 possibilities, it is standard bridged (it's l1 exists) or it is a native token (only l2, no bridge)
+    // - for standard bridged l2, extract its l1 and add it. simple. (addValidBridgeableToken(l1Address, l2Address))
+    // - for l2-only token, we cant extract l1, but we add it to the list for better UX with a flag `isL2OnlyToken`
+
+    const lowercasedErc20L1orL2Address = erc20L1orL2Address.toLowerCase()
+
+    // check if the address can be found on  L1
+    const l1TokenDetails = await getERC20TokenDetails({
+      walletAddress,
+      erc20L1orL2Address,
+      provider: l1.provider
+    })
+    if (l1TokenDetails) {
+      const l1Address = lowercasedErc20L1orL2Address
+      const l2Address = await getL2ERC20Address({
+        erc20L1Address: lowercasedErc20L1orL2Address,
+        l1Provider: l1.provider,
+        l2Provider: l2.provider
+      }) // if this fails, then L2-only token
```

Comment:
If this fails, wouldn't it mean the address exists on L1 but not on L2? So `L1-only token`?

Reply:
Also, if this fails, is it going to throw or just return `undefined`? We aren't checking for `!l2Address` or catching an error, is this correct?